### PR TITLE
Update dotnet-kill to use cross-platform 'ps' and 'kill' commands

### PR DESCRIPTION
**Pull request by brandongregoryscott**
Fixes issue #63 

Add 'ps' module for wrapping 'fkill' and 'ps-list' packages, update dotnet-kill module to call ps.list and ps.kill, add associated tests

Tested on Mac OS 10.15.5, Ubuntu 18.04.3, and Windows 10 Pro Build 18363.836

### CODE REVIEW

File: _modules/dotnet-kill.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+// #endregion Imports
+
+// -----------------------------------------------------------------------------------------
+// #region Functions
+// -----------------------------------------------------------------------------------------
+
+const dotnetKill = {
+    cmd() {
+        return {
```
Comment (wintondeshong) [inline _modules/dotnet-kill.js:18]: Not suggesting we change this now, but if this "pattern" results in adding more properties/methods than those here, I feel we need to extract a 'command/cmd' object type/class

File: _modules/ps.test.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+            // Assert
+            expect(shellExitSpy).toHaveBeenCalledWith(1);
+        });
+
+        test("when psList succeeds, it returns an array of processes", async () => {
+            // Arrange
+            const mockPid       = faker.random.number();
+            const mockProcesses = [
+                { pid: mockPid },
+                { pid: mockPid + 1 },
```
Comment (wintondeshong) [inline _modules/ps.test.js:87]: If faker randomizes maximum number, the `+ 1` will cause an exception

File: _modules/ps.test.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        test("when given a process name, it returns an array of processes that match", async () => {
+            // Arrange
+            const mockPid           = faker.random.number();
+            const expectedProcess   = {
+                name: "dotnet",
+                pid:  mockPid
+            };
+            const unexpectedProcess = {
+                name: "paint",
+                pid:  mockPid + 1
```
Comment (wintondeshong) [inline _modules/ps.test.js:107]: Same as above

**Comment by brandongregoryscott** [comment]
Thanks for taking a look @wintondeshong! I'll test this out again. I could've sworn it was killing my dotnet instance even when it was being run thru the `and-cli`. I'll double check to see what's going on there before I merge anything - admittedly I was testing this more on my Mac, since that's where the problem initially arose, but I'm seeing the same behavior as you on Windows.

File: _modules/dotnet-kill.js (hunk 1 above)
Comment (brandongregoryscott) [reply _modules/dotnet-kill.js:18]: Not a bad idea. I like it a lot more than doing the string building/splitting manually, but I could see this manifesting itself in some sort of builder or type.

File: _modules/ps.test.js (hunk 2 above)
Comment (brandongregoryscott) [reply _modules/ps.test.js:87]: Good catch!

**Comment by brandongregoryscott** [comment]
So weird. I'm testing it again on Windows, without any code changes - and it's working as expected. I thought maybe it was the difference of calling the `and-cli-dev` alias to spawn `dotnet` vs. my current globally installed version `and-cli`, and that wasn't it. Thought maybe it was due to having the VS Code terminal open as administrator, or Cmder open as administrator, and that doesn't seem to be the case. I'm going to keep using it for a few more days (preferring the `dotnet -k` vs. manually killing dotnet with ctrl+c) to see if it acts up, but I'm not sure how I got into that state before.

Do you know what version of the `and-cli` you were running `dotnet` with? Was it the dev alias, the global version, or a project version?

**Comment by wintondeshong** [comment]
Just did a fresh clone of your branch, install, using alias (I had set up an for your fork :P and-cli-brandongregoryscott). It worked as expected 🤷‍♂️ Not sure what changed. I'll also try to use it to stop my dotnet processes I might use. Honestly, if nothing crops up in the next day or so I'd say this is good to go.

**Pull request merged**